transport/peer/hostport/peer.go

```diff
@@ -0,0 +1,73 @@
+package hostport
+
+import (
+	"go.uber.org/yarpc/transport"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewPeerIdentifier creates a new hostport.PeerIdentifier
+func NewPeerIdentifier(hostport string) PeerIdentifier {
+	return PeerIdentifier(hostport)
+}
```

Comment:
You can drop the constructor if the type alias is public anyway. Users don't
get any benefit from calling `NewPeerIdentifier("foo")` instead of
`PeerIdentifier("foo")` unless we're doing some validation or parsing.